### ISSUE TITLE
[Bug]: Provide the ability to view leaderboard of another user when logged out

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#5754]
### Summary

The app had been providing the ability to view another user's leaderboard through the Explore section. Specifically, when you click on an image in the "Explore" screen and tap on the "View user page" menu it opens up the uploader's profile/leaderboard and contributions. This is possible when the user has logged in. When the user is logged out, we show the "View user page" option but the same does not work as it should. We should explore the feasibility of providing the ability to view the user's leaderboard information even when the user of the app has not logged in.

### Steps to reproduce

1. Open the app when and make sure you are not logged in
2. Click on an image from the "Explore" screen
3. Tap on the overflow menu and click on the "View user page" option

### Expected behaviour

The user profile information and the leaderboard information are shown clearly

### Actual behaviour

We get a weird error saying our login token expired and ask us to login again. Check the following recording for an idea: 

### Android version

a

### Commons app version

5.0.1

### Screen-shots

Check this screen recording here:

### Would you like to work on the issue?

None

**Comment by whym** [comment]
If the expected behavior is show an in-app profile page (that contains leaderboard information etc), we might want to call it something like "user's profile" instead of "user page". "User page" sounds ambiguous with on-wiki user page, which is written by the user and not automatically generated. (for example, https://commons.wikimedia.org/wiki/User:Whym )

Or alternatively, could we change the expected behavior instead, and show (a preview of) the on-wiki user page and keep the "user page" label?

(I would suggest preview, in case the user page is too long.)

**Comment by sivaraam** [comment]
> If the expected behavior is show an in-app profile page (that contains leaderboard information etc), we might want to call it something like "user's profile" instead of "user page".

Makes sense! Calling it a user profile does sound better.

> "User page" sounds ambiguous with on-wiki user page, which is written by the user and not automatically generated. (for example, https://commons.wikimedia.org/wiki/User:Whym )
> 
> Or alternatively, could we change the expected behavior instead, and show (a preview of) the on-wiki user page and keep the "user page" label?
> 

Rather than that would it make sense to show the achievements screen itself but also provide a way for them access the user page from there?

**Comment by whym** [comment]
Yeah, showing achivements etc and having a link to the user page somewhere in the same screen  (if one exists, not all users have user pages) sound great.

**Comment by sivaraam** [comment]
> If the expected behavior is show an in-app profile page (that contains leaderboard information etc), we might want to call it something like "user's profile" instead of "user page".

For now, I've done this change alone in the following PR: #5771

**Comment by prototyperspective** [comment]
The error shown is "Session expired. Please log in again" and it pops up twice. One can only view a user page at second try, the first time nothing happens except that this error shows.

**Comment by Kota-Jagadeesh** [comment]
hi @sivaraam, Can i take up the issue?
